pilot/pkg/model/config.go

```diff
@@ -528,6 +528,7 @@ var (
 // ResolveHostname uses metadata information to resolve a service reference to
 // a fully qualified hostname. The metadata namespace and domain are used as
 // fallback values to fill up the complete name.
+// TODO: Deprecated. Get rid of this method
```

Comment:
Can you merge and the subsequent method? The next one is a bit naive since it does not take the optional namespace int account. There is some function I wrote that you can transfer here.

Reply:
This is taking the namespace into account. Istio service had two fields for short name and long name. Ideally, this code should just move out and they should simply give us a FQDN.